### PR TITLE
Move mod_env in its own class and load it when required

### PR DESCRIPTION
**Pull request by sathieu**

### CODE REVIEW

**Comment by sathieu** [comment]
The tests are probably failing for unrelated reason (connection error).

File: manifests/vhost.pp [hunk 1]
```diff
@@ -539,11 +539,19 @@
...
+  $use_env_mod = $setenv and ! empty($setenv)
+  if ($use_env_mod) {
+    if ! defined(Class['apache::mod::env']) {
+      include ::apache::mod::env
+    }
+  }
   # Check if mod_setenvif is required and not yet loaded.
   # create an expression to simplify the conditional check
-  $use_setenv_mod = ($setenv and ! empty($setenv)) or ($setenvif and ! empty($setenvif)) or ($setenvifnocase and ! empty($setenvifnocase))
+  $use_setenvif_mod = ($setenvif and ! empty($setenvif)) or ($setenvifnocase and ! empty($setenvifnocase))
```
Comment (bmjen) [inline manifests/vhost.pp]: @sathieu  This appears to be a change in behavior. Previously, the `use_setenv_if_mod` would only be `true` if `use_setenv_mod` were true. Now it is evaluated independently and below... can add the config fragment without the setting of `$setenv`.

Is this desired behavior?

File: spec/acceptance/default_mods_spec.rb [hunk 2]
```diff
@@ -34,12 +34,18 @@ class { 'apache':
           default_mods => false,
         }
         apache::vhost { 'defaults.example.com':
-          docroot => '#{$doc_root}/defaults',
-          aliases => {
+          docroot     => '#{$doc_root}/defaults',
```
Comment (bmjen) [inline spec/acceptance/default_mods_spec.rb]: indentation is wrong here.

**Comment by bmjen** [comment]
Ping @sathieu. I'm hoping to cut a release today and would like to include this PR.

File: manifests/vhost.pp (hunk 1 above)
Comment (sathieu) [inline manifests/vhost.pp]: Previously mod_setenvif was loaded when only setenv was used, which is not needed at all.

The new behavior handle the modules independently (see [mod_env](http://httpd.apache.org/docs/2.4/mod/mod_env.html) and [mod_setenvif](http://httpd.apache.org/docs/2.4/mod/mod_setenvif.html)).

So yes: behavior desired.

File: spec/acceptance/default_mods_spec.rb (hunk 2 above)
Comment (sathieu) [inline spec/acceptance/default_mods_spec.rb]: Fixed

**Pull request merged**